test/jdk/javax/swing/JFileChooser/6798062/bug6798062.java

```diff
@@ -21,54 +21,87 @@
  * questions.
  */
 
-/* @test %W% %E%
-   @bug 6798062
-   @summary Memory Leak on using getFiles of FileSystemView
-   @author Pavel Porvatov
-   @modules java.desktop/sun.awt
-            java.desktop/sun.awt.shell
-   @run applet/manual=done bug6798062.html
-*/
+/*
+ * @test
+ * @bug 6798062
+ * @requires (os.family == "windows")
+ * @summary Memory Leak on using getFiles of FileSystemView
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @modules java.desktop/sun.awt
+ *          java.desktop/sun.awt.shell
+ * @run main/manual bug6798062
+ */
 
 import sun.awt.OSInfo;
 import sun.awt.shell.ShellFolder;
 
-import javax.swing.*;
+import java.awt.Dimension;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.awt.*;
 import java.io.File;
 import java.io.FileNotFoundException;
-
-public class bug6798062 extends JApplet {
-
-    private final JSlider slider = new JSlider(0, 100);
-
-    private final JTextField tfLink = new JTextField();
-
-    private final JButton btnStart = new JButton("Start");
-
-    private final JButton btnStop = new JButton("Stop");
-
-    private final JButton btnGC = new JButton("Run System.gc()");
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+public class bug6798062 {
+
+    private static final String instructionsText = """
```

Comment:
`instructionsText` may be capitalize as using capital letter is common convention for final variables.

Reply:
That is for constant, IIRC...

Reply:
Either way is fine… And `instructionsText` *is* a constant but it's not public.